### PR TITLE
Add govuk-repo-mirror IAM user

### PR DESCRIPTION
**Pull request by steventux**
https://trello.com/c/TX8kPxzK/319-start-backing-up-our-code-in-aws-codecommit-not-gitlab

This IAM user will mirror GOVUK Github repos on AWS Code Commit.
They require permissions to list, create and push to repositories.

Not sure if all the required changes are included in this PR, `terraform plan` runs fine but due to the nature of the changes I can't fully test this.

### CODE REVIEW

File: terraform/projects/infra-govuk-repo-mirror/main.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+variable "stackname" {
+  type        = "string"
+  description = "Stackname"
+}
+
+resource "aws_iam_user" "govuk_code_commit_user" {
+  name = "govuk-${var.aws_environment}-govuk-code-commit-user"
+}
+
+resource "aws_iam_policy" "govuk_code_commit_user" {
```
Comment (boffbowsh) [inline terraform/projects/infra-govuk-repo-mirror/main.tf]: I think this policy also needs attaching to the user

Comment (steventux) [reply terraform/projects/infra-govuk-repo-mirror/main.tf]: @boffbowsh I took this setup from a terraform example I found here: https://github.com/cloudspinners/terraform-aws-codecommit-repository/blob/master/committer_group.tf I'm not sure the `govuk_code_commit_user` policy is necessary if we're already attaching the managed policy `policy/AWSCodeCommitPowerUser` ...?

Comment (boffbowsh) [reply terraform/projects/infra-govuk-repo-mirror/main.tf]: Yeah I think you're right, you can delete this block if you're using the managed policy.

Comment (steventux) [reply terraform/projects/infra-govuk-repo-mirror/main.tf]: @boffbowsh updated https://github.com/alphagov/govuk-aws/pull/593/commits/ec3124866367caf8252a5ce05f23924ee3df4764

**Pull request merged**